hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/migration/AddRootContractIdMigrationTest.java

```diff
@@ -0,0 +1,167 @@
+package com.hedera.mirror.importer.migration;
+
+/*-
+ * ‌
+ * Hedera Mirror Node
+ * ​
+ * Copyright (C) 2019 - 2021 Hedera Hashgraph, LLC
+ * ​
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.io.File;
+import java.util.Arrays;
+import java.util.List;
+import javax.annotation.Resource;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import org.apache.commons.io.FileUtils;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.jdbc.core.BeanPropertyRowMapper;
+import org.springframework.jdbc.core.JdbcOperations;
+import org.springframework.test.context.TestPropertySource;
+
+import com.hedera.mirror.importer.EnabledIfV1;
+import com.hedera.mirror.importer.IntegrationTest;
+
+@EnabledIfV1
+@Tag("migration")
+@TestPropertySource(properties = "spring.flyway.target=1.50.2")
+class AddRootContractIdMigrationTest extends IntegrationTest {
+
+    @Resource
+    private JdbcOperations jdbcOperations;
+
+    @Value("classpath:db/migration/v1/V1.51.0__contract_logs_root_id.sql")
+    private File migrationSql;
+
+    @BeforeEach
+    void setup() {
+        revertMigration();
+    }
+
+    @Test
+    void verifyRootContractIdMigrationEmpty() throws Exception {
+        migrate();
+        assertThat(retrieveContractLogs()).isEmpty();
+    }
+
+    @Test
+    void verifyRootContractIdMigration() throws Exception {
+
+        persistContractResult(Arrays.asList(
+                contractResult(1, 1L),
+                contractResult(2, 2L),
+                contractResult(3, null)
+        ));
+        persistContractLog(Arrays.asList(
+                contractLog(1, 1, 0),
+                contractLog(1, 2, 1),
+                contractLog(1, 3, 2),
+                contractLog(2, 1, 0),
+                contractLog(3, 1, 0)
+        ));
+        // migration
+        migrate();
+
+        assertThat(retrieveContractLogs())
+                .hasSize(5).extracting(MigrationContractLog::getContractId)
+                .containsExactlyInAnyOrder(1L, 1L, 1L, 2L, 3L);
```

Comment:
Order is important since the query has ordering
```suggestion
                .containsExactly(1L, 1L, 1L, 2L, 3L);
```

Reply:
Yep, that actually caught an error in the test.  Fixed both.